/*
* Brian R Taylor
* brian.taylor@bolderflight.com
* 
* Copyright (c) 2020 Bolder Flight Systems
*/

/*
* This protobuf file describes the data log format, which is transpiled
* to a C header and source file for use in the flight code. There should
* only be a single message defined, DatalogMessage, which defines all
* of the data to be logged. The message structure should be flat
* (i.e. no nested messages), consisting of native types.
*/

syntax = "proto3";

/* Data log message */
message DatalogMessage {
  double time_s = 1;
  repeated float imu_accel_mps2 = 2;
  repeated float imu_gyro_radps = 3;
  repeated float imu_mag_ut = 4;
  bool gnss_updated = 5;
  int32 gnss_utc_year = 6;
  int32 gnss_utc_month = 7;
  int32 gnss_utc_day = 8;
  int32 gnss_utc_hour = 9;
  int32 gnss_utc_min = 10;
  float gnss_utc_sec = 11;
  int32 gnss_fix = 12;
  int32 gnss_num_sv = 13;
  repeated float gnss_ned_vel_mps = 14;
  double gnss_lat_rad = 15;
  double gnss_lon_rad = 16;
  float gnss_alt_m = 17;
  repeated float ins_accel_mps2 = 18;
  repeated float ins_gyro_radps = 19;
  repeated float ins_ned_vel_mps = 20;
  double ins_lat_rad = 21;
  double ins_lon_rad = 22;
  float ins_alt_m = 23;
  float ins_pitch_rad = 24;
  float ins_roll_rad = 25;
  float ins_yaw_rad = 26;
  float fmu_static_press_pa = 27;
  float static_press_pa = 28;
  float diff_press_pa = 29;
  float filt_static_press_pa = 30;
  float filt_diff_press_pa = 31;
  float press_alt_m = 32;
  float agl_alt_m = 33;
  float ias_mps = 34;
  float eas_mps = 35;
  float input_voltage = 36;
  float regulated_voltage = 37;
  float pwm_voltage = 38;
  float sbus_voltage = 39;
}

